<?xml version="1.0" encoding="UTF-8"?>
<launch>   
    <arg name="fcu_url" default="/dev/ttyUSB1:921600" /> <!-- Yusuf please check this -->
    <arg name="gcs_url" default="udp://@192.168.0.192" /> <!-- Yusuf please check this -->
    
    <!-- TFs -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="transform_body2bl" args="-0.13 0 -0.069 0 0 0 1 body base_link" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="transform_sensor2bl" args="-0.13 0 -0.069 0 0 0 1 os_sensor base_link" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="transform_odom2camera_init" args="0 0 0 0 0 0 1 camera_init odom" />

    <!-- LIDAR -->
    <arg name="lidar_freq" default="10" />
    <include file="$(find livox_ros_driver2)/launch_ROS1/msg_MID360.launch">
        <arg name="publish_freq" value="$(arg lidar_freq)" />
    </include>

    <include file="$(find fast_lio)/launch/mapping_avia.launch"/>

    <!-- PIXHAWK -->
     <rosparam>
        /mavros/vision_pose/tf/listen: true
        /mavros/vision_pose/tf/frame_id: camera_init
        /mavros/vision_pose/tf/child_frame_id: base_link
        /mavros/vision_pose/tf/rate_limit: 100
    </rosparam>
    <arg name="tgt_system" default="1" />
    <arg name="tgt_component" default="1" />
    <arg name="log_output" default="screen" />
    <include file="$(find mavros)/launch/node.launch">
        <arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
        <arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />
        <arg name="fcu_url" value="$(arg fcu_url)" />
        <arg name="gcs_url" value="$(arg gcs_url)" />
        <arg name="tgt_system" value="$(arg tgt_system)" />
        <arg name="tgt_component" value="$(arg tgt_component)" />
        <arg name="log_output" value="$(arg log_output)" />
    </include>
    <rosparam>
        /mavros/vision_pose/tf/listen: true
        /mavros/vision_pose/tf/frame_id: camera_init
        /mavros/vision_pose/tf/child_frame_id: base_link
        /mavros/vision_pose/tf/rate_limit: 100
    </rosparam>

    <!-- DELTA -->
    <include file="$(find delta_2)/launch/stabilise.launch">
        <arg name="sim" value="false"/>
    </include>

    <!-- SETPOINT PUBLISHER -->
    <node name="demo_publisher_1" pkg="rostopic" type="rostopic" args="pub /manipulator/state std_msgs/String 'STAB_delta' -r 10" />
    <node name="demo_publisher_2" pkg="rostopic" type="rostopic" args="pub /tooltip_setpoint/pose geometry_msgs/PoseStamped '{header: {seq: 0, stamp: 'now', frame_id: 'camera_init'}, pose: {position: {x: 0.5, y: 0.0, z: 0.0}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}}}' -r 100 -s" />
</launch>
